shardy/dialect/sdy/ir/ops.td

```diff
@@ -446,6 +453,10 @@ def Sdy_AllGatherOp : Sdy_Op<"all_gather",
     %2 = sdy.all_gather gathering_axes=[{"b", "c"}, {}, {"d"}\] %1 to_sharding=<@mesh, [{"a"}, {}, {}\]> : tensor<8x8xf32>
     ```
 
+    **Constraints:**
+    - `MeshAttr` of result and operand is the same
```

Comment:
Lets rephrase to - both operand and result shardings should be bound to the same `MeshAttr`